packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -313,23 +315,64 @@ const ProductDetail = () => {
     const handleProductBundleAddToCart = async (variant, selectedQuantity) => {
         try {
             const childProductSelections = Object.values(childProductSelection)
-            const bundledProductItems = childProductSelections.map((child) => {
-                return {
-                    productId: child.variant.productId,
-                    quantity: child.quantity
-                }
-            })
 
             const productItems = [
                 {
                     productId: product.id,
                     price: product.price,
                     quantity: selectedQuantity,
-                    bundledProductItems
+                    bundledProductItems: childProductSelections.map((child) => {
+                        return {
+                            productId: child.variant.productId,
+                            quantity: child.quantity
+                        }
+                    })
                 }
             ]
 
-            await addItemToNewOrExistingBasket(productItems)
+            const res = await addItemToNewOrExistingBasket(productItems)
```

Comment:
May be we can move the logic to a helper class. If possible move common code between cart and product-detail in a function so that we can reuse the same code